client/state/data-layer/wpcom/read/following/mine/index.js

```diff
@@ -86,29 +90,53 @@ export function requestPage( store, action, next ) {
 }
 
 const MAX_PAGES_TO_FETCH = MAX_ITEMS / ITEMS_PER_PAGE;
+
 export function receivePage( store, action, next, apiResponse ) {
 	if ( ! isValidApiResponse( apiResponse, action ) ) {
 		receiveError( store, action, next, apiResponse );
 		return;
 	}
 
 	const { page, number } = apiResponse;
-
+	const follows = subscriptionsFromApi( apiResponse );
 	store.dispatch(
 		receiveFollowsAction( {
-			follows: subscriptionsFromApi( apiResponse ),
+			follows,
 			totalCount: apiResponse.total_subscriptions,
 		} )
 	);
 
+	forEach( follows, ( follow ) => {
+		seenSubscriptions.add( follow.feed_URL );
+	} );
+
 	// Fetch the next page of subscriptions where applicable
 	if ( number > 0 && page <= MAX_PAGES_TO_FETCH && isSyncingFollows() ) {
 		store.dispatch( requestPageAction( page + 1 ) );
 		return;
 	}
+	// all done syncing
+	// diff what we saw vs. what's in state and remove anything extra
+	const currentFollows = getReaderFollows( store.getState() );
+	const followsToRemove = filter( currentFollows, ( follow ) => follow.ID &&
```

Comment:
ID is to validate that we have a valid subscription object, not something pending or inflight. is_following could be dropped, it was just to avoid sending an unfollow for something we don't follow.